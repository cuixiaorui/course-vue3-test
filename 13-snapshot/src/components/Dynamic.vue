<template>
  <div>
    <div v-if="!showB">A</div>
    <div v-else>B</div>
  </div>
</template>

<script>
export default {
  props: ["showB"]
};
</script>

<style></style>
