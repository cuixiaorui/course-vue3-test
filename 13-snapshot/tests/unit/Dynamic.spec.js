import Dynamic from "@/components/Dynamic.vue";
import { shallowMount } from "@vue/test-utils";

describe("Dynamic", () => {
  test("dynamic snapshot default ", () => {
    const wrapper = shallowMount(Dynamic, {});

    expect(wrapper.element).toMatchSnapshot();
  });
  test("dynamic snapshot true", () => {
    const wrapper = shallowMount(Dynamic, {
      props: {
        showB: true
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
