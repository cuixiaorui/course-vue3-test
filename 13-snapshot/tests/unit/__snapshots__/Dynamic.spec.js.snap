// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dynamic dynamic snapshot default  1`] = `
<div>
  <div>
    A
  </div>
</div>
`;

exports[`Dynamic dynamic snapshot true 1`] = `
<div>
  <div>
    B
  </div>
</div>
`;
